### PR TITLE
Refine ChordStructure schema

### PR DESCRIPTION
**Pull request by davidforrest**
I'm adding the various chord structures that we'll need for non-diatonic mode, and writing simple tests.

### CODE REVIEW

File: src/tests/ChordStructure.test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { ChordType } from '../generator/ChordType'
 
 test('triad chord type returns four chord structures', () => {
   const expected = new Set([
-  ChordStructure.MAJOR,
+    ChordStructure.MAJOR,
```
Comment (jsbean) [inline src/tests/ChordStructure.test.js]: 🙌

**Comment by jsbean** [comment]
@davidforrest, I've got an idea: what if we merged this PR in as-is. This first push is helpful because it fleshes out the model for the pre-existing chord types.

You could then create a new branch off of `generator-v2`, which has the CI testing fixed, to add the new cases in with the new schema?

File: src/generator/ChordStructure.js [hunk 2]
```diff
@@ -5,93 +5,147 @@ import { ChordType } from "./ChordType"
 export const ChordStructure = {
   MAJOR: {
     displayName: "M",
-    mode: Mode.LYDIAN_DOMINANT,
+    modeConstructor: Mode.MAJOR, // TODO: David to define all modeConstructors in mode.js
+    possibleModeEnvironments: ["Major", "minor"], // FIXME: It would feel safer to derive possibleModeEnvironments based on array lengths >0 in commonRootOffsets.
```
Comment (jsbean) [inline src/generator/ChordStructure.js]: Ah, this is a great idea 👍

**Comment by davidforrest** [comment]
@jsbean sounds good to me!

**Comment by jsbean** [comment]
Cool! I am actually going to make a PR real quick that brings in the fixed CI testing, and I might update the tests so that we aren't looking for a "neat"-politan sixth yet. So hold your horses 🏇 .

**Comment by davidforrest** [comment]
i wonder if i reply to these comments by email if it works?

**Comment by davidforrest** [comment]
aha yes it does!

**Comment by jsbean** [comment]
> i wonder if i reply to these comments by email if it works?
> […](#)

Oh, you can also change your notification settings so you don't get an email for every annoying comment I leave here !

**Comment by davidforrest** [comment]
then how would i know when all the annoying things are happening!

**Comment by jsbean** [comment]
🤯

**Pull request merged**